twilight-lavalink/src/node.rs

```diff
@@ -539,6 +548,78 @@ impl Connection {
         Ok(())
     }
 
+    async fn get_outgoing_endpoint_based_on_event(
+        &self,
+        outgoing: &OutgoingEvent,
+    ) -> Result<(Method, hyper::Uri), NodeError> {
+        let address = self.config.address;
+        tracing::debug!("forwarding event to {}: {outgoing:?}", address,);
+
+        let guild_id = OutgoingEvent::guild_id(outgoing);
+        let no_replace = OutgoingEvent::no_replace(outgoing);
+
+        if let Some(session) = self.lavalink_session_id.lock().await.clone() {
+            tracing::debug!(
+                "Found session id {}. Generating the url and method for event type.",
+                session
+            );
+
+            if let OutgoingEvent::Destroy(_) = outgoing.clone() {
+                let destroy_uri = Uri::builder()
+                    .scheme("http")
+                    .authority(address.to_string())
+                    .path_and_query(format!("/v4/sessions/{session}/players/{guild_id}"))
+                    .build()
+                    .unwrap();
+                return Ok((Method::DELETE, destroy_uri));
+            }
+            let uri = Uri::builder()
+                .scheme("http")
+                .authority(address.to_string())
+                .path_and_query(format!(
+                    "/v4/sessions/{session}/players/{guild_id}?noReplace={no_replace}"
+                ))
+                .build()
+                .unwrap();
+            return Ok((Method::PATCH, uri));
```

Comment:
This can be simplified
```suggestion
            let mut path = format!("/v4/sessions/{session}/players/{guild_id}");
            if matches!(outgoing, OutgoingEvent::Destroy(_)) {
                path.push_str(&format!("?noReplace={no_replace}"));
            }
            let uri = Uri::builder()
                .scheme("http")
                .authority(address.to_string())
                .path_and_query(path)
                .build()
                .expect("uri is valid");
            return if matches!(outgoing, OutgoingEvent::Destroy(_)) {
                Ok((Method::DELETE, uri))
            } else {
                Ok((Method::PATCH, uri))
            };
```